aks/postgres/resources.tf

```diff
@@ -180,6 +180,17 @@ resource "azurerm_storage_container" "backup" {
   container_access_type = "private"
 }
 
+resource "azurerm_storage_container_immutability_policy" "backup" {
+  count = local.azure_enable_backup_storage && var.environment == "production" ? 1 : 0
```

Comment:
@RMcVelia Is there any chance this behaviour could be controlled from a variable? which itself defaults to this policy?

on NPQ we'd like to have all backups be immutable to prevent accidental overwrites?

Reply:
The plan was to use immutability for production backup storage accounts only, but also change the postgres backup action to not overwrite (which would cover all environments). Hopefully that would fit what you are after?